pyne/source_sampling.pyx

```diff
@@ -478,14 +652,185 @@ cdef class Sampler:
+        None 
+
+        Returns
+        -------
+        cell_number : int
+        
+        """
+        cdef int cell_number
+        cell_number = (<cpp_source_sampling.Sampler *> self._inst).get_src_cell_number()
+        return cell_number
+
+    def get_src_xxx(self):
+        """get_src_xxx(self)
+        Get the source particle x coordinate
+        
+        Parameters
+        ----------
+        None 
+
+        Returns
+        -------
+        xxx : double
+        
+        """
+        cdef double xxx
+        xxx = (<cpp_source_sampling.Sampler *> self._inst).get_src_xxx()
+        return xxx
+
+    def get_src_yyy(self):
+        """get_src_yyy(self)
+        Get the source particle y coordinate
+        
+        Parameters
+        ----------
+        None 
+
+        Returns
+        -------
+        yyy : double
+        
+        """
+        cdef double yyy
+        yyy = (<cpp_source_sampling.Sampler *> self._inst).get_src_yyy()
+        return yyy
+
+    def get_src_zzz(self):
+        """get_src_zzz(self)
+        Get the source particle z coordinate
+        
+        Parameters
+        ----------
+        None 
+
+        Returns
+        -------
+        zzz : double
+        
+        """
+        cdef double zzz
+        zzz = (<cpp_source_sampling.Sampler *> self._inst).get_src_zzz()
+        return zzz
+   
+    def get_src_erg(self):
+        """get_src_erg(self)
+        Get the source particle energy
+        
+        Parameters
+        ----------
+        None 
+
+        Returns
+        -------
+        erg : double
+        
+        """
+        cdef double erg
+        erg = (<cpp_source_sampling.Sampler *> self._inst).get_src_erg()
+        return erg
+   
+    def get_src_wgt(self):
+        """get_src_wgt(self)
+        Get the source particle weight
+        
+        Parameters
+        ----------
+        None 
+
+        Returns
+        -------
+        wgt : double
+        
+        """
+        cdef double wgt
+        wgt = (<cpp_source_sampling.Sampler *> self._inst).get_src_wgt()
+        return wgt
+   
+    def get_src_xyzew(self):
+        """get_src_xyzew(self)
+        Get source particle xxx, yyy, zzz, erg and wgt
+        
+        Parameters
+        ----------
+        None
+        
+        Returns
+        -------
+        res1 : std::vector< double >
+        
+        """
+        cdef cpp_vector[double] rtnval
+        cdef np.npy_intp rtnval_proxy_shape[1]
+        rtnval = (<cpp_source_sampling.Sampler *> self._inst).get_src_xyzew()
         rtnval_proxy_shape[0] = <np.npy_intp> rtnval.size()
         rtnval_proxy = np.PyArray_SimpleNewFromData(1, rtnval_proxy_shape, np.NPY_FLOAT64, &rtnval[0])
         rtnval_proxy = np.PyArray_Copy(rtnval_proxy)
         return rtnval_proxy
     
+
+cdef class SourceParticle:
+    """Constructor for class SourceParticle
+    
+    Attributes
+    ----------
+    zzz (double) : The z coordinate of the source particle
```

Comment:
Are there some attributes missing here perhaps?

I'm also wondering about the naming of these... why `zzz`? Rather than simply `z`?